exercises/change/example_gen.go

```diff
@@ -38,25 +38,59 @@ type OneCase struct {
 	Expected    interface{}
 }
 
+func (c OneCase) Valid() bool {
+	valid, _ := determineExpected(c.Expected)
+	return valid
+}
+
+func (c OneCase) IntSlice() []int {
+	_, list := determineExpected(c.Expected)
+	return list
+}
+
+// determineExpected examines an .Expected interface{} object and determines
+// whether an error is indicated by an int value (actually -1) in the JSON,
+// or whether .Expected is a slice of integer coins.
+func determineExpected(expected interface{}) (valid bool, list []int) {
+	_, ok := expected.(int)
+	if ok {
+		return false, nil
+	}
+	ilist, ok := expected.([]interface{})
+	if ok {
```

Comment:
optional: could make this an `if !ok { return false, nil }`, then many lines of code can be unindented